qadence_protocols/types.py

```diff
@@ -1,6 +1,11 @@
 from __future__ import annotations
 
 from enum import Enum
+from typing import Union
+
+from torch import Tensor
+
+MeasurementData = Union[Tensor, list, tuple, None]
```

Comment:
```suggestion
MeasurementData = Tensor | list | tuple | None
```

For a reason, I would leave `None` out.

Reply:
Still have to use Union for python 3.9

Reply:
Eventually, `MeasurementData` becomes a dataclass